### ISSUE TITLE
Selenium binds too many sockets

### ISSUE THREAD

**Issue by songsb13** [SeleniumHQ/selenium#3772]
## Meta -
OS:  Windows 7 Pro, Windows 10 Pro

Selenium Version: 3.3.1(python)

Browser:  Chrome 57.0.2987.133 (64-bit) (Chrome driver 2.29)

Browser Version:  

## Expected Behavior -

I tried to test with multiple proxies; therefore, I had to run multiple chrome drivers.

## Actual Behavior -

It binds socket more than 3,000(per chrome driver) and raises errors such as no buffer space available, address already in use and other network errors related with socket.

## Steps to reproduce -

run multiple chrome drivers with multiprocessing 

sample code:

```
from selenium import webdriver
from multiprocessing import Process, freeze_support
import time

def run_selenium(driver_path, proxy):
    chrome_options = webdriver.ChromeOptions()
    chrome_options.add_argument("--incognito")
    chrome_options.add_argument("--proxy-server={}".format(proxy))

    driver = webdriver.Chrome(driver_path, chrome_options=chrome_options)
    driver.get('www.youtubue.com')
    driver.quit()


if __name__ == '__main__':
    freeze_support()
    proxy_list = open('proxy_list.txt').readlines()
    for proxy in proxy_list:
        p = Process(target=run_selenium, args=('./chromedriver.exe', proxy))
        p.start()
        time.sleep(5)
```

**Comment by cgoldberg** [comment]
> run multiple chrome drivers with multiprocessing

can you be a **lot** more descriptive?

**Comment by songsb13** [comment]
@cgoldberg added sample code :)

**Comment by ariasuni** [comment]
I think I have the same problem with Firefox under Linux. I use selenium 2.53.* with scrapy launching many tasks in parallel, and when upgrading to 3.*, I have errors like those:

```
  File "/usr/lib/python2.7/site-packages/twisted/internet/tcp.py", line 546, in doConnect
    err = self.socket.getsockopt(socket.SOL_SOCKET, socket.SO_ERROR)
  File "/usr/lib/python2.7/socket.py", line 228, in meth
    return getattr(self._sock,name)(*args)
socket.error: [Errno 9] Bad file descriptor
```

and

```
  File "/home/ariasuni/projets/crawling_v2/.venvpy2/lib/python2.7/site-packages/twisted/internet/abstract.py", line 446, in startWriting
    self.reactor.addWriter(self)
  File "/home/ariasuni/projets/crawling_v2/.venvpy2/lib/python2.7/site-packages/twisted/internet/epollreactor.py", line 127, in addWriter
    EPOLLOUT, EPOLLIN)
  File "/home/ariasuni/projets/crawling_v2/.venvpy2/lib/python2.7/site-packages/twisted/internet/epollreactor.py", line 94, in _add
    self._poller.modify(fd, flags)
exceptions.IOError: [Errno 9] Bad file descriptor
```

**Comment by barancev** [comment]
Duplicate of #5758
